File: app/src/main/java/com/example/runtimetracker/RunCursorAdapter.java
package com.example.runtimetracker;

import android.annotation.SuppressLint;
import android.content.Context;
import android.database.Cursor;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CursorAdapter;
import android.widget.TextView;

import java.text.SimpleDateFormat;
import java.util.Date;

public class RunCursorAdapter extends CursorAdapter {
    private final LayoutInflater layoutInflater;

    public RunCursorAdapter(Context context, Cursor c, int flags) {
        super(context, c, flags);
        layoutInflater = LayoutInflater.from(context);
    }

    @Override
    public View newView(Context context, Cursor cursor, ViewGroup viewGroup) {
        return layoutInflater.inflate(R.layout.run_record, viewGroup, false);
    }

    @Override
    public void bindView(View view, Context context, Cursor cursor) {
        setDate(view, cursor);
        setTime(view, cursor);
        setDistance(view, cursor);
    }

    private void setDate(View view, Cursor cursor) {
        @SuppressLint("Range") long milli = cursor.getLong(cursor.getColumnIndex("Run_Date"));
        Date date = new Date(milli);
        SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy");
        ((TextView) view.findViewById(R.id.txt_run_date)).setText(formatter.format(date));
    }

    private void setTime(View view, Cursor cursor) {
        @SuppressLint("Range") String runTime = cursor.getString(cursor.getColumnIndex("Run_Time"));
        ((TextView) view.findViewById(R.id.txt_run_time)).setText(runTime);
    }

    private void setDistance(View view, Cursor cursor) {
        @SuppressLint("Range") String runDist = cursor.getString(cursor.getColumnIndex("Run_Distance"));
        ((TextView) view.findViewById(R.id.txt_run_dist)).setText(runDist+" Km");
    }
}

File: app/src/main/java/com/example/runtimetracker/RunCursorAdapter.kt
package com.example.runtimetracker

import android.annotation.SuppressLint
import android.content.Context
import android.database.Cursor
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CursorAdapter
import android.widget.TextView
import java.text.SimpleDateFormat
import java.util.*

class RunCursorAdapter(context: Context?, c: Cursor?, flags: Int) : CursorAdapter(context, c, flags) {
    private val layoutInflater: LayoutInflater

    override fun newView(context: Context, cursor: Cursor, viewGroup: ViewGroup): View {
        return layoutInflater.inflate(R.layout.run_record, viewGroup, false)
    }

    override fun bindView(view: View, context: Context, cursor: Cursor) {
        setDate(view, cursor)
        setTime(view, cursor)
        setDistance(view, cursor)
    }

    private fun setDate(view: View, cursor: Cursor) {
        @SuppressLint("Range") val milli = cursor.getLong(cursor.getColumnIndex("Run_Date"))
        val date = Date(milli)
        val formatter = SimpleDateFormat("dd-MM-yyyy")
        (view.findViewById<View>(R.id.txt_run_date) as TextView).text = formatter.format(date)
    }

    private fun setTime(view: View, cursor: Cursor) {
        @SuppressLint("Range") val runTime = cursor.getString(cursor.getColumnIndex("Run_Time"))
        (view.findViewById<View>(R.id.txt_run_time) as TextView).text = runTime
    }

    private fun setDistance(view: View, cursor: Cursor) {
        @SuppressLint("Range") val runDist = cursor.getString(cursor.getColumnIndex("Run_Distance"))
        (view.findViewById<View>(R.id.txt_run_dist) as TextView).text = "$runDist Km"
    }

    init {
        layoutInflater = LayoutInflater.from(context)
    }
}